bin/test/i18n/test_i18n_metrics.rb

```diff
@@ -0,0 +1,19 @@
+require_relative '../test_helper'
+require_relative '../../i18n/i18n_metrics'
+require 'cdo/aws/metrics'
+
+class I18nMetricsTest < Minitest::Test
+  def setup
+    Cdo::Metrics.client = Aws::CloudWatch::Client.new(stub_responses: true)
+    I18n::Metrics.stubs(:machine_id).returns('local_machine')
+  end
+
+  def expect_metric(name, value, dimensions)
+    Cdo::Metrics.expects(:put_metric).with("I18n", {metric_name: name, value: value, dimensions: dimensions})
+  end
+
+  def test_report_runtime
+    expect_metric(:Runtime, 1, [{name: "dim1", value: 1}, {name: "dim2", value: 2}, {name: "Environment", value: :test}, {name: "MachineId", value: "local_machine"}])
+    I18n::Metrics.report_runtime([{name: "dim1", value: 1}, {name: "dim2", value: 2}]) {1}
```

Comment:
not sure if I'm crazy.. but I would expect the metrics to be called and then expected? (aka these two lines in the other order)

Reply:
me too!!! and then artem told me i was wrong and he was right

Reply:
`Mocha#expects` works as an `assertion` check in this case: https://github.com/freerange/mocha#mock-objects

Reply:
It should be this way because it's kind of you need to assign a spy by method (add `.expects`) before the execution